src/components/il-tabs/il-tabs.component.js

```diff
@@ -25,42 +25,62 @@ class TabComponent extends LitElement {
     this.compact = this.offsetWidth < TabComponent.compactSizePixelWidth;
   }
 
-   getLinkElement() {
-     return document.getElementById(this.linkedby);
+   addPolyfillForContainerSupprt() {
+    console.debug("Tab: No support for @container detected: using manual resize");
+    window.addEventListener('load', this.handleResize.bind(this));
+    window.addEventListener('resize', this.handleResize.bind(this));
+   }
+
+   removePolyfillForContainerSupprt() {
+    console.debug("Tab: removing resize");
+    window.removeEventListener('load', this.handleResize.bind(this));
+    window.removeEventListener('resize', this.handleResize.bind(this));
    }
 
   constructor() {
     super();
+  }   
+
+  connectedCallback() {
+   super.connectedCallback();
+   if (!TabComponent.hasContainerSupport()) {
+    this.addPolyfillForContainerSupprt();
+  }
+}
+  
+  disconnectedCallback() {
     if (!TabComponent.hasContainerSupport()) {
-      console.debug("Tab: No support for @container detected: using manual resize");
-      window.addEventListener('load', this.handleResize.bind(this));
-      window.addEventListener('resize', this.handleResize.bind(this));
+      this.removePolyfillForContainerSupprt();
     }
-    let firstItem = true;
-    this.getAllTabs().forEach(item => {
-      item.addEventListener('keydown', this.keyboard);
+    super.disconnectedCallback();
+  }
+
+  getAllTabs() {
+    return Array.from(this.querySelector('*[slot=tabs]').children);
+  }
+
+  getAllPanels() {
+    return Array.from(this.querySelectorAll('*:not([slot])'));
+  }
+
+  setTabs() {
+    console.debug("Tab: set information from tabs slot");
+    this.getAllTabs().forEach((item, i) => {
+      item.addEventListener('keydown', this.handleKeypress);
       item.addEventListener('click', e => this.setActivePanel(item));
       item.setAttribute('role', 'tab');
-      if (firstItem) {
+      if (i == 0) {
         item.setAttribute('tabindex', 0);
         item.setAttribute('aria-selected', 'true');
         this.setActivePanel(item);
-        firstItem = false;
       } else {
         item.setAttribute('tabindex', -1);
       }
```

Comment:
Can these be a separate class method, initTab() or something more descriptive of what it's doing?